### PR TITLE
User Update

### LINKED ISSUE #18: User Update
As a user,
I can click edit link and change attributes,
where the app sends information
```
{
    "first_name": "Update First",
    "last_name": "Update Last",
    "phone_number": "3333333333"
    "email": "Name@email.com", 
    "api_key": asD7fsdfwef2332!%sdf
}
``` 
to `PATCH /api/v1/user/edit`,
which finds the user,
then updates user information,
which returns the response
```
status: 202
body:
{
  "data":
  {
    "type": "user",
    "id": "1",
    "attributes":
    {
      "first_name": "Update First",
      "last_name": "Update Last",
      "phone_number": "3333333333",
      "email": "Name@email.com",
      "api_key": "asD7fsdfwef2332!%sdf"
    }
  }
}
```
### Checks
- [x] Checks to see if user already exists in user table
- [x] For success, returns status code 202
- [x] Returns correct response format json
- [x] Add endpoint documentation to README.md

Sad path
- [x] Unable to find user, returns status code 404 with message: "Failed to find user"
- [x] Unable to update user, returns status code 422 with message: "Failed to update user"

### PR DESCRIPTION
**Pull request by lpile**
## Changes proposed in this pull request:
* Add user update spec
* Add #create action to UsersController with route
* Remove :password from validates_presence_of in User model for update functionality
* Add edit user endpoints to README.md

## Card(s) closed in this pull request:
close #18 

## What did you struggle on to complete?
* Nothing at this moment.

### Test Coverage Percentage: 100%
- [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
- [x] Checked coverage/index.html - did not add any new code that's not covered by testing (if possible)
- [x] Ran the test suite - all tests are passing
- [x] Merged in the latest master to my branch with git pull origin master & resolved merge conflicts
- [x] I have commented my code, particularly in hard-to-understand areas

## Helpful Resources:
* [ActiveRecord #create docs](https://guides.rubyonrails.org/active_record_basics.html#create)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,8 +39,67 @@ body:
...
 {
-  "error"=>"Failed to login"
+  "error"=>"Failed to register"
+}
+```
+
+#### Edit a single user
+**Request:**
+```
+POST /api/v1/user/edit
```
Comment (annekemcgrady) [inline README.md]:
```suggestion
PATCH /api/v1/user/edit
```

Comment (lpile) [reply README.md]: Good catch!
